code/drasil-lang/lib/Language/Drasil.hs

```diff
@@ -281,10 +282,21 @@ module Language.Drasil (
   -- * Type Synonyms
   , ConstQDef, SimpleQDef, ModelQDef
   , PExpr
+
+  -- * Type systems
```

Comment:
I like the idea of exporting it so that external libraries can build their own type systems too, but there's no hard requirement to export anything other than 'check', 'inferFromContext', and 'Typed(..)'.

I think I should actually remove allOfType, it's more of a 'utility' function anything.

Reply:
Export a minimal list for now, and re-export them later when it's actually needed.